### PR TITLE
feat(permissions): add use-is-authorized hook

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This pull request adds a `useIsAuthorized()` hook. It should be pretty self explanatory from the specs.

### CODE REVIEW

File: packages/application-shell-connectors/src/components/application-context/index.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export {
+  Context,
```
Comment (tdeekens) [inline packages/application-shell-connectors/src/components/application-context/index.js]: The React docs state a useContext call needs the return of `createContext`. Not only the consumer.

File: packages/permissions/src/hooks/use-is-authorized/use-is-authorized.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+      const isAuthorized = shouldMatchSomePermissions
+        ? hasSomePermissions(demandedPermissions, actualPermissions)
+        : hasEveryPermissions(demandedPermissions, actualPermissions);
+
+      return isAuthorized;
+    }
+  : () => {
+      throw new Error(
+        `React hooks do not seem to be available yet in the installed React version "${
```
Comment (jonnybel) [inline packages/permissions/src/hooks/use-is-authorized/use-is-authorized.js]: 🆒

Comment (tdeekens) [reply packages/permissions/src/hooks/use-is-authorized/use-is-authorized.js]: We do that on other hooks too. I didn't care to test it again. In another hook there is a teset.

**Comment by tdeekens** [comment]
Feel free to merge when deemed good.

**Pull request merged**